packages/core/src/components/portal/_portal.scss

```diff
@@ -34,7 +34,3 @@ The children of a `Portal` component are appended to the `<body>` element.
 
 Styleguide components.portal.js
 */
-
-.pt-portal {
-  display: inline-block;
-}
```

Comment:
I don't see how this does that your PR describes. This removes this rule, so that .pt-portal elements aren't forced to be inline-block, but that doesn't mean pt-portal elements won't be inline (span) or inline-block (input, button).